### PR TITLE
`Reskin` |  DAO Hierarchy level indication

### LINKED ISSUE #1720: DAO Hierarchy level indication
8) @xraystyle1980 / @nicolaus-sherrill  for the dao hierarchy, I currently visually have no way of knowing (beyond the title) which level of the hierarchy I am currently in. Compare this to the original below with the strong underline. Is this intended functionality? If so, I'd question it especially for Myosin DAO who has three levels
Screenshot 2024-05-06 at 13 05 04
Screenshot 2024-05-06 at 13 35 58

_Originally posted by @tomstuart123 in https://github.com/decentdao/fractal-interface/issues/1578#issuecomment-2096395374_

### PR DESCRIPTION
**Pull request by Da-Colon**
**Description:**

- Implemented the new Figma design to clearly indicate the currently viewed DAO.

**Notes:**

1. **Code Cleanup:**
   - The `NodeInfoCard` contains a significant amount of commented-out code related to the old formatting, which can likely be removed for clarity.

2. **Component Update:**
   - The DAO Info Card on the DAO Dashboard and Hierarchy pages, which previously used the same components, are now quite different in the new design (e.g., no gear icon, reduced icon size). We should decouple these components and create a separate component for the Hierarchy page rather than modifying the existing one.

Hierarchy Page

---

**Additional Information:**

- This branch is based on my GraphQL bug fix branch to allow for proper testing and visibility.

Closes #1720

### CODE REVIEW

**Comment by xraystyle1980** [comment]
@Da-Colon updated the [Hierarchy handoff to include a prototype](https://www.figma.com/proto/9hUYtvCKjOyB92vt8ukYtD/Product-Handoff?page-id=122%3A13462&node-id=2746-5942&viewport=1037%2C-346%2C0.35&t=57SJEp8vmUSy3SkE-1&scaling=scale-down&starting-point-node-id=2746%3A5942&show-proto-sidebar=1) to show clicking on the full card should switch organizations. 

*Also includes hover and active states for the hierarchy cards. 

@Da-Colon I added the three dots icon-button to these designs. I can't recall if we removed them intentionally or they just didn't make it to the reskin design. @tomstuart123 can you weigh in here?

**Comment by Da-Colon** [comment]
> @Da-Colon updated the [Hierarchy handoff to include a prototype](https://www.figma.com/proto/9hUYtvCKjOyB92vt8ukYtD/Product-Handoff?page-id=122%3A13462&node-id=2746-5942&viewport=1037%2C-346%2C0.35&t=57SJEp8vmUSy3SkE-1&scaling=scale-down&starting-point-node-id=2746%3A5942&show-proto-sidebar=1) to show clicking on the full card should switch organizations.
> 
> *Also includes hover and active states for the hierarchy cards.
> 
> @Da-Colon I added the three dots icon-button to these designs. I can't recall if we removed them intentionally or they just didn't make it to the reskin design. @tomstuart123 can you weigh in here?

I'm going to take this comment and make a new Issue. Between Adam's PR #1766 and the work I saw him doing earlier today. There will be alot of code changes coming for the `DAOInfoCard` which I'll need to touch to adjust the settings icon and the hover/active states. 

We can talk on Monday, but I think it might be better to decouple the use of this component between two pages and have separate components. This will make this alot easier to follow code-wise I believe. 

But I believe this PR covers the single issue indication of the Current DAO.

File: src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx [hunk 1]
```diff
@@ -68,31 +72,42 @@ export function DaoHierarchyNode({
...
     >
-      <DAONodeInfoCard
-        parentAddress={parentAddress}
-        node={fractalNode}
-        freezeGuard={daoData?.freezeGuard}
-        guardContracts={daoData?.freezeGuardContracts}
-      />
+      <Box
+        border={currenDAOAddress === getAddress(daoAddress || '') ? '4px solid' : 'none'}
+        borderColor={currenDAOAddress === getAddress(daoAddress || '') ? 'neutral-4' : 'none'}
```
Comment (adamgall) [inline src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx:77]: `getAddress` will throw if you give it the empty string. Don't do this plz

File: src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx [hunk 2]
```diff
@@ -68,31 +72,42 @@ export function DaoHierarchyNode({
...
+          freezeGuard={daoData?.freezeGuard}
+          guardContracts={daoData?.freezeGuardContracts}
+        />
+      </Box>
 
       {/* CHILD NODES */}
       {fractalNode?.nodeHierarchy.childNodes.map(childNode => (
-        <HStack
+        <Flex
+          flexDirection="row"
```
Comment (adamgall) [inline src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx:90]: Row is default don't need to specify it

File: src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx [hunk 3]
```diff
@@ -25,6 +26,9 @@ export function DaoHierarchyNode({
   daoAddress?: Address;
   depth: number;
 }) {
+  const {
+    node: { daoAddress: currenDAOAddress },
```
Comment (adamgall) [inline src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx:30]: Nit: typo here: `currentDAOAddress`

File: src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx [hunk 4]
```diff
@@ -68,31 +72,42 @@ export function DaoHierarchyNode({
...
           gap="1.25rem"
         >
           <Icon
             as={ArrowElbowDownRight}
+            my={`${NODE_HEIGHT_REM / 2.5}rem`}
+            ml="0.5rem"
             boxSize="32px"
-            color="neutral-6"
+            color={
+              currenDAOAddress === getAddress(childNode.daoAddress || '') ? 'celery-0' : 'neutral-6'
```
Comment (adamgall) [inline src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx:101]: No empty string into `getAddress` plz

**Review by adamgall — CHANGES_REQUESTED** [review]

File: src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx (hunk 4 above)
Comment (Da-Colon) [reply src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx:101]: o dang didn't know that. I'll get this fixed then

**Review by DarksightKellar — CHANGES_REQUESTED** [review]
Changes LGTM, but should the border not have a radius on it? There's a visible gap:

Screenshot 2024-05-13 at 12 42 22

**Review by xraystyle1980 — CHANGES_REQUESTED** [review]
Border radius on active card is missing

**Comment by tomstuart123** [comment]
Extra request - how hard would it be to only include the border once the DAO has loaded?

Reason - see below. The border jumps around for a bit as the DAO in the hierarchy load animation forces it to reverberate
Screenshot 2024-05-13 at 15 59 40

link to try it - https://deploy-preview-1767.app.dev.fractalframework.xyz/hierarchy?dao=sep:0x065D37327801F5a0c01AD7CfB388288eb2378fFf

**Comment by Da-Colon** [comment]
Good catch @DarksightKellar 

localhost_3000_hierarchy_dao=sep_0x35b798b19F70b236b9Dcb156F21F4f0f4Ca44C49(Desktop)


@tomstuart123 I was able to make it happen

Node loading

**Review by xraystyle1980 — DISMISSED** [review]
The Favorite button is super tiny. And can you make the entire card clickable? [See prototype](https://www.figma.com/proto/9hUYtvCKjOyB92vt8ukYtD/Product-Handoff?page-id=122%3A13462&node-id=2746-5942&viewport=719%2C-376%2C0.32&t=6P9F7fu4rIVHJ5X8-1&scaling=scale-down&starting-point-node-id=2746%3A5942&show-proto-sidebar=1)

**Comment by tomstuart123** [comment]
@Da-Colon - that looks fire, Thanks so much

@xraystyle1980 from my understanding this is linked to @Da-Colon 's comment above. As this component and DAO home have the same PR, the tiny Favourite's icon is shared between them. This should get resolved by an existing PR on the tiny 'Favourites' on dao home. 

Is that correct devs?

**Comment by Da-Colon** [comment]
> @Da-Colon - that looks fire, Thanks so much
> 
> @xraystyle1980 from my understanding this is linked to @Da-Colon 's comment above. As this component and DAO home have the same PR, the tiny Favourite's icon is shared between them. This should get resolved by an existing PR on the tiny 'Favourites' on dao home.
> 
> Is that correct devs?



> @Da-Colon - that looks fire, Thanks so much
> 
> @xraystyle1980 from my understanding this is linked to @Da-Colon 's comment above. As this component and DAO home have the same PR, the tiny Favourite's icon is shared between them. This should get resolved by an existing PR on the tiny 'Favourites' on dao home.
> 
> Is that correct devs?

Yep thats correct. @xraystyle1980 To another point. I think the two components have been separated already and only have shared components (such as the `FavoriteIcon` component. Which I think is why the full click isn't working. I'll fix the whole linking in the follow PR for issue #1773

**Review by DarksightKellar — CHANGES_REQUESTED** [review]
Lol sorry to be that guy, but there's still a gap when you look closely 😅
Screenshot 2024-05-14 at 15 54 53

I think I've had this problem before on a component that was too small to notice, where even matching radii didn't seem to work visually.

**Comment by Da-Colon** [comment]
> Lol sorry to be that guy, but there's still a gap when you look closely 😅 Screenshot 2024-05-14 at 15 54 53
> 
> I think I've had this problem before on a component that was too small to notice, where even matching radii didn't seem to work visually.
> 
> EDIT: Hmm, maybe reducing the radius on the card itself so the border occludes the curve (only on active state) might do the trick? I have a feeling it'll lead to unwanted jank tho

ah blah, good eye. yeah I need to just move this all down into the DAONodeInfoCard, which would be the plan. anything else I'd do here would just be hacky. Can we get this merged in as is and I work on cleaning this up in the next PR?

**Review by DarksightKellar — APPROVED** [review]

**Pull request merged**